src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -631,6 +633,7 @@ BuildJobQuery(MultiNode *multiNode, List *dependedJobList)
 	jobQuery->groupClause = groupClauseList;
 	jobQuery->limitOffset = limitOffset;
 	jobQuery->limitCount = limitCount;
+	jobQuery->havingQual = havingQual;
```

Comment:
> can such a situation exist with no aggregates on the targetList, but an aggregate on having clause ? In this case hasAggs determination needs to take havingQual into consideration too.

We added aggregates to target list in WorkerExtendedOpNode(), so hasAggs should be fine here.

And we have regression tests for this case;
https://github.com/citusdata/citus/blob/having_support/src/test/regress/sql/multi_simple_queries.sql#L180